src/adapter/src/notice.rs

```diff
@@ -342,10 +345,22 @@ impl fmt::Display for AdapterNotice {
             AdapterNotice::WebhookSourceCreated { url } => {
                 write!(f, "URL to POST data is '{url}'")
             }
+            AdapterNotice::DroppedInUseIndex(DroppedInUseIndex {
+                index_name,
+                dependant_objects,
+            }) => {
+                write!(f, "The dropped index {index_name} is being used by the following objects: {}. The index will be dropped from the catalog, but it will continue to be maintained and take up resources!", separated(", ", dependant_objects))
```

Comment:
Should we also say something about the fact that the index will cease to exist on next restart?

Reply:
Yes, we were thinking with Alex to not emphasize the maintenance window restart too much, because it's just an incidental thing, e.g., it might not occur if there is no release on a certain week. Also, we don't want users waiting for the maintenance window with the replanning, because then if the index not being used anymore slows things down, then we'll get confused whether it was the version upgrade or just an index being dropped.

But I'm planning to open a follow-up PR with adding docs for these notices, and there I will probably mention this anyway.

Reply:
I think it's important for the user to be aware that at some point in the future, outside of their control, the objects using this index will become less efficient as they will no longer have access to the index. We don't need to specifically mention maintenance windows, I guess.

Reply:
Hmm, because maybe a user would want to rely on this dropped index being used, and then get surprised when it is cleaned up after a maintenance window? Well, I _can_ imagine this happening, unfortunately. What a mess... 

I'm trying to think up a new wording.

Reply:
We also discussed having a dedicated page where the notice hints can point to that will provide more context.

Reply:
> Well, I can imagine this happening, unfortunately.

This seems like an anti-pattern. Until we implement plan pinning as a feature that is enabled by default for every installed materialized view and index dataflow, users should not rely on their plans being stable across restarts.

Reply:
I agree that users should not be relying on this, but unfortunately I think some _will_ rely on it unless we explicitly teach them not to.

Reply:
Ok, so what should be the wording? This is the current wording:

"The dropped index {index_name} is being used by the following objects: {}. The index will be dropped from the catalog, but it will continue to be maintained and take up resources!"

we could instead have

"The dropped index {index_name} is being used by the following objects (until they are dropped or Materialize is restarted): {}. The index is now dropped from the catalog, but it will continue to be maintained and take up resources!"

What do you all think?

Reply:
I suggest 

> until they are dropped, altered or Materialize is restarted

in order to also cover #20725.

Reply:
I'm opening a PR with the following:

"The dropped index {index_name} is being used by the following objects: {}. The index is now dropped from the catalog, but it will continue to be maintained and take up resources until all dependent objects are dropped, altered, or Materialize is restarted!"